### PR TITLE
CI Add GitHub workflow for creating releases

### PR DESCRIPTION
**Pull request by BenjaminBossan**
Solves #34 

Minor additions:

- Added some documentation around releases and developing skops.
- Fix a typo (sklearn -> skops)
- Make test dependencies installable

Note: For obvious reasons, I haven't tested yet whether the GH action really works.

### CODE REVIEW

**Comment by BenjaminBossan** [comment]
I'm getting this error:

> requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://huggingface.co/api/models/skops/test-4a551f91-bec7-4996-9e9f-ed75ecb0395c/commit/main (Request ID: xg-VOMqxP3BFWz4qxgtbk)

([link](https://github.com/skops-dev/skops/runs/7408257939?check_suite_focus=true))

Any idea what this could be about? It seems to be unrelated to my changes.

**Review by adrinjalali — APPROVED** [review]
Nice!

**Comment by adrinjalali** [comment]
Thos 400 errors are because the backend is unstable and sometimes gives 400. We just re-run the tests for them to pass, and they almost always do with max 3 retries. We can also use a retry decorator for our tests.

File: README.rst [hunk 1]
```diff
@@ -17,3 +17,39 @@ Demo a model using HuggingFace Spaces
...
+==============================
+
+Follow these steps if you want to contribute to the skops development.
+
+Using conda
+-----------
+
+.. code:: bash
+
+          conda create -n skops python=3.10
```
[the remark this answers is not in the document]
Comment (BenjaminBossan) [reply README.rst:34]: I can make that change, but could you please explain  what the advantage is?

[the remark this answers is not in the document]
Comment (adrinjalali) [reply README.rst:34]: I've seen odd python behaviors with the one shipped on the main channel. Easier to only deal with the one on conda-forge which is maintained by the community.

[the remark this answers is not in the document]
Comment (BenjaminBossan) [reply README.rst:34]: Really? How can something so fundamental create odd behavior >_<;; Anyway, I'll make the change

**Comment by adrinjalali** [comment]
The error on the CI doesn't seem like the usual we get. Would you what's causing it @SBrandeis?

```
                except Exception as exc:
                    # Raise the first exception encountered
>                   raise RuntimeError(
                        f"Error while uploading {operation.path_in_repo} to the Hub"
                    ) from exc
E                   RuntimeError: Error while uploading model.pkl to the Hub
/opt/hostedtoolcache/Python/3.8.13/x64/lib/python3.8/site-packages/huggingface_hub/_commit_api.py:263: RuntimeError
```

**Review by adrinjalali — APPROVED** [review]

File: .github/workflows/publish-pypi.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      with:
+        python-version: '3.x'
+
+    - name: Install dependencies
+      run: |
+        python -m pip install -U pip
+        python -m pip install -U setuptools wheel twine
+
+    - name: Publish package to TestPyPI
+      uses: pypa/gh-action-pypi-publish@v1.5.0
```
Comment (adrinjalali) [inline .github/workflows/publish-pypi.yml:30]: I just noticed this action only publishes the built packages. We need to build them before uploading them.

Comment (BenjaminBossan) [reply .github/workflows/publish-pypi.yml:34]: I should have checked, my bad. I added a build step above.

File: .github/workflows/publish-pypi.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        python-version: '3.x'
+
+    - name: Install dependencies
+      run: |
+        python -m pip install -U pip
+        python -m pip install -U setuptools wheel twine build
+
+    - name: Generate distribution archives
+      run: |
+        python -m build
```
Comment (adrinjalali) [inline .github/workflows/publish-pypi.yml:31]: oh, didn't know this does the thing.

**Review by adrinjalali — APPROVED** [review]
Let's see how it goes.

File: .github/workflows/publish-pypi.yml (hunk 3 above)
Comment (BenjaminBossan) [reply .github/workflows/publish-pypi.yml:31]: Same, it was new to me. But I figured if the [docs](https://packaging.python.org/en/latest/tutorials/packaging-projects/#generating-distribution-archives) recommend this way, it is best practice.

**Pull request merged**